pkg/controller/persistentpodstate/persistent_pod_state_controller.go

```diff
@@ -0,0 +1,448 @@
+/*
+Copyright 2021 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package persistentpodstate
+
+import (
+	"context"
+	"flag"
+	"reflect"
+	"strconv"
+	"strings"
+
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+	appsv1beta1 "github.com/openkruise/kruise/apis/apps/v1beta1"
+	"github.com/openkruise/kruise/pkg/util"
+	"github.com/openkruise/kruise/pkg/util/controllerfinder"
+	"github.com/openkruise/kruise/pkg/util/ratelimiter"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+	utilpointer "k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+func init() {
+	flag.IntVar(&concurrentReconciles, "persistentpodstate-workers", concurrentReconciles, "Max concurrent workers for PersistentPodState controller.")
+}
+
+var (
+	concurrentReconciles = 3
+
+	// kubernetes
+	KindSts = appsv1.SchemeGroupVersion.WithKind("StatefulSet")
+	// kruise
+	KruiseKindSts = appsv1beta1.SchemeGroupVersion.WithKind("StatefulSet")
+
+	// PersistentPodStateNodeTopology default generation persistent the same node topology
+	PersistentPodStateNodeTopology      = "kruise.io/persistent-node-topology"
```

Comment:
Is this better?
```yaml
    kruise.io/auto-generate-persistent-pod-state: true
    # optional
    kruise.io/required-persistent-topology: topology.kubernetes.io/zone[,xxx]
    # optional
    kruise.io/preferred-persistent-topology: kubernetes.io/hostname[,xxx]
```

Also the annotations/labels that provide for users to configure should be defined in `apis/apps/v1alpha1/persistent_pod_state_types.go` instead of controller. Files in apis will be synchronized to kruise-api for users.